### PR TITLE
Feature/dvad 3350 add marriage cypress tests

### PR DESCRIPTION
**Pull request by aarongill1**

### CODE REVIEW

File: cypress/fixtures/marriage/expected-marriage-record.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+const search = {
+  systemNumber: '',
+  forenames: 'Tester',
+  surname: 'SOLO',
+  day: '29',
```
Comment (JamesWoodwardHMPO) [inline cypress/fixtures/marriage/expected-marriage-record.js:7]: Would prefer the date to look like this

dom: {
  day: '29',
  month: '02',
  year: '2012'
}

and change subsequent usage accordingly.

File: cypress/fixtures/marriage/expected-marriage-records.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
+'use strict';
+
+const search = {
+  systemNumber: '',
+  forenames: 'Tester',
+  surname: 'MULTIPLE',
+  day: '29',
```
Comment (JamesWoodwardHMPO) [inline cypress/fixtures/marriage/expected-marriage-records.js:7]: Would prefer the date to look like this

dom: {
day: '29',
month: '02',
year: '2012'
}

and change subsequent usage accordingly.

File: cypress/fixtures/marriage/expected-no-records.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+const search = {
+  systemNumber: '',
+  forenames: 'Winston',
+  surname: 'Churchil',
+  day: '30',
```
Comment (JamesWoodwardHMPO) [inline cypress/fixtures/marriage/expected-no-records.js:7]: Would prefer the date to look like this

dom: {
day: '29',
month: '02',
year: '2012'
}

and change subsequent usage accordingly.

File: cypress/integration/acceptance/marriage/marriage-details.specs.js [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    it('returns me to the search page', () => {
+      MarriageSearchPage.shouldBeVisible();
+    });
+
+    it('has empty form values', () => {
+      MarriageSearchPage.hasExpectedValues({
+        systemNumber: '',
+        surname: '',
+        forenames: '',
+        'dom-day': '',
```
Comment (JamesWoodwardHMPO) [inline cypress/integration/acceptance/marriage/marriage-details.specs.js:91]: Change inline with above

File: cypress/integration/acceptance/marriage/marriage-details.specs.js [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+      });
+    });
+  });
+
+  describe('When I select the "Edit search" link on the results page', () => {
+    const search = {
+      systemNumber: '',
+      surname: 'NotRealPersonSurname',
+      forenames: 'NotRealPersonForename',
+      day: 1,
```
Comment (JamesWoodwardHMPO) [inline cypress/integration/acceptance/marriage/marriage-details.specs.js:104]: Change inline with above

File: cypress/integration/acceptance/marriage/marriage-results.specs.js [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+const expectedMultipleRecords = require('../../../fixtures/marriage/expected-marriage-records');
+
+describe('Marriage results', () => {
+  before(() => {
+    LoginPage.login();
+  });
+
+  describe('When I perform a search that returns no records', () => {
+    const searchNoRecords = {
+      search: { surname: 'InvalidRecord', forenames: 'Test', day: '01', month: '01', year: '2011' },
```
Comment (JamesWoodwardHMPO) [inline cypress/integration/acceptance/marriage/marriage-results.specs.js:17]: Change inline with above

File: cypress/integration/acceptance/marriage/marriage-results.specs.js [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+    describe('When I select the "New search" link on the results page', () => {
+      before(() => {
+        MarriageSearchPage.visit();
+        MarriageSearchPage.performSearch(searchMultipleRecords.search);
+        MarriageResultsPage.clickNewSearchButton();
+      });
+
+      it('a search page should be displayed with no values', () => {
+        MarriageSearchPage.shouldBeVisible();
+        MarriageSearchPage.hasExpectedValues({ systemNumber: '', surname: '', forenames: '', dobd: { day: '', month: '', year: '' } });
```
Comment (JamesWoodwardHMPO) [inline cypress/integration/acceptance/marriage/marriage-results.specs.js:102]: Should be dom not dobd

File: cypress/integration/acceptance/marriage/marriage-search.specs.js [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+        it('displays an error message, requests a valid dom', () => {
+          MarriageSearchPage.invalidDOMDay();
+        });
+      });
+      describe('invalid month', () => {
+        before(() => {
+          MarriageSearchPage.visit();
+          MarriageSearchPage.shouldBeVisible();
+          MarriageSearchPage.performSearch({
+            surname: 'TEST', forenames: 'TEST', day: '01', month: 'XX', year: '2010'
```
Comment (JamesWoodwardHMPO) [inline cypress/integration/acceptance/marriage/marriage-search.specs.js:188]: Change inline with above

**Review by JamesWoodwardHMPO — CHANGES_REQUESTED** [review]

File: cypress/fixtures/marriage/expected-marriage-record.js (hunk 1 above)
Comment (aarongill1) [reply cypress/fixtures/marriage/expected-marriage-record.js:7]: Done

File: cypress/fixtures/marriage/expected-marriage-records.js (hunk 2 above)
Comment (aarongill1) [reply cypress/fixtures/marriage/expected-marriage-records.js:7]: Done

File: cypress/fixtures/marriage/expected-no-records.js (hunk 3 above)
Comment (aarongill1) [reply cypress/fixtures/marriage/expected-no-records.js:7]: Done

File: cypress/integration/acceptance/marriage/marriage-details.specs.js (hunk 4 above)
Comment (aarongill1) [reply cypress/integration/acceptance/marriage/marriage-details.specs.js:91]: Done

File: cypress/integration/acceptance/marriage/marriage-details.specs.js (hunk 5 above)
Comment (aarongill1) [reply cypress/integration/acceptance/marriage/marriage-details.specs.js:104]: Done

File: cypress/integration/acceptance/marriage/marriage-results.specs.js (hunk 7 above)
Comment (aarongill1) [reply cypress/integration/acceptance/marriage/marriage-results.specs.js:102]: Done

File: cypress/integration/acceptance/marriage/marriage-search.specs.js [hunk 9]
```diff
@@ -0,0 +1,222 @@
...
+        });
+      });
+    });
+
+    describe('with a missing first name', () => {
+      before(() => {
+        MarriageSearchPage.visit();
+        MarriageSearchPage.performSearch({
+          surname: 'Surname',
+          day: 5,
```
[the remark this answers is not in the document]
Comment (aarongill1) [reply cypress/integration/acceptance/marriage/marriage-search.specs.js:133]: Done

File: cypress/integration/acceptance/marriage/marriage-search.specs.js [hunk 10]
```diff
@@ -0,0 +1,222 @@
...
+
+      it('requests a forename', () => {
+        MarriageSearchPage.hasErrorMessage('Please enter at least one forename');
+      });
+
+      describe('and a missing surname', () => {
+        before(() => {
+          MarriageSearchPage.visit();
+          MarriageSearchPage.performSearch({
+            day: 5,
```
[the remark this answers is not in the document]
Comment (aarongill1) [reply cypress/integration/acceptance/marriage/marriage-search.specs.js:151]: Done

File: cypress/integration/acceptance/marriage/marriage-search.specs.js [hunk 11]
```diff
@@ -0,0 +1,222 @@
...
+        });
+      });
+    });
+    describe('with an invalid date of marriage that has an', () => {
+      describe('invalid day', () => {
+        before(() => {
+          MarriageSearchPage.visit();
+          MarriageSearchPage.shouldBeVisible();
+          MarriageSearchPage.performSearch({
+            surname: 'TEST', forenames: 'TEST', day: 'XX', month: '10', year: '2010'
```
[the remark this answers is not in the document]
Comment (aarongill1) [reply cypress/integration/acceptance/marriage/marriage-search.specs.js:176]: Done

File: cypress/integration/acceptance/marriage/marriage-search.specs.js [hunk 12]
```diff
@@ -0,0 +1,222 @@
...
+          MarriageSearchPage.invalidDOMYear();
+        });
+      });
+      describe('a date in the future', () => {
+        const dateToday = DateTime.now().plus({ days: 1 });
+        before(() => {
+          MarriageSearchPage.visit();
+          MarriageSearchPage.performSearch({
+            surname: 'McFly', forenames: 'Marty Jr',
+            day: dateToday.day, month: dateToday.month, year: dateToday.year
```
[the remark this answers is not in the document]
Comment (aarongill1) [reply cypress/integration/acceptance/marriage/marriage-search.specs.js:213]: Done

File: cypress/pages/marriage/MarriageDetailsPage.js [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+
+  /**
+   * Check marriage registrations details page has the expected data
+   */
+  static hasLimitedRecord(record) {
+    const { groom, bride, registrar } = record;
+    const rows = [
+      `System number ${record.id}`,
+      `Date of marriage ${record.dateOfMarriage}`,
+      `Place of marriage ${record.placeOfMarriage.address}`,
```
[the remark this answers is not in the document]
Comment (aarongill1) [reply cypress/pages/marriage/MarriageDetailsPage.js:45]: Amended along with fixture records

File: cypress/pages/marriage/MarriageDetailsPage.js [hunk 14]
```diff
@@ -0,0 +1,139 @@
...
+
+  /**
+   * Check marriage registrations details page has the expected data
+   */
+  static hasCompleteRecord(record) {
+    const { groom, bride, fatherOfGroom, fatherOfBride, registrar } = record;
+    const rows = [
+      `System number ${record.id}`,
+      `Date of marriage ${record.dateOfMarriage}`,
+      `Place of marriage ${record.placeOfMarriage.address}`,
```
[the remark this answers is not in the document]
Comment (aarongill1) [reply cypress/pages/marriage/MarriageDetailsPage.js:73]: Done

File: cypress/pages/marriage/MarriageResultsPage.js [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+  }
+
+  /**
+   * Check expected title is displayed
+   *
+   * @param expected
+   */
+  static hasExpectedTitle(expected) {
+    const { search, results } = expected;
+    const title = `${results.length === 0 ? 'No' : results.length} records found for ${search.forenames} ${search.surname} ${search.day}/${search.month}/${search.year}`;
```
[the remark this answers is not in the document]
Comment (aarongill1) [reply cypress/pages/marriage/MarriageResultsPage.js:22]: Done

File: cypress/pages/marriage/MarriageSearchPage.js [hunk 16]
```diff
@@ -0,0 +1,115 @@
...
+
+  /**
+   * Perform a marriage registration search with the given params
+   *
+   * @param systemNumber
+   * @param surname
+   * @param forenames
+   * @param dom
+   */
+  static performSearch({
```
[the remark this answers is not in the document]
Comment (aarongill1) [reply cypress/pages/marriage/MarriageSearchPage.js:39]: Done

[the remark this answers is not in the document]
Comment (aarongill1) [reply cypress/pages/marriage/MarriageSearchPage.js:39]: Done

File: cypress/pages/marriage/MarriageSearchPage.js [hunk 17]
```diff
@@ -0,0 +1,115 @@
...
+   */
+  static hasExpectedValues({
+    systemNumber,
+    surname,
+    forenames,
+  }) {
+    cy.get('#system-number').should('have.value', systemNumber);
+    cy.get('#surname').should('have.value', surname);
+    cy.get('#forenames').should('have.value', forenames);
+    cy.get('label[for=dom-day]').contains('Day');
```
[the remark this answers is not in the document]
Comment (aarongill1) [reply cypress/pages/marriage/MarriageSearchPage.js:79]: Done

File: cypress/pages/DetailsPage.js [hunk 18]
```diff
@@ -37,7 +37,7 @@ class DetailsPage extends BackToSearchPage {
    * @param rows
    */
   static hasExpectedRows(rows) {
-    cy.get('table.details tr').each((element, index) => {
+    cy.get('table.details').each((element, index) => {
```
[the remark this answers is not in the document]
Comment (aarongill1) [reply cypress/pages/DetailsPage.js:40]: Have reverted - was required because of a bug in the marriage details page template that has been fixed.

**Review by JamesWoodwardHMPO — APPROVED** [review]
Approved, just need to merge the changes from master.

**Review by JamesWoodwardHMPO — APPROVED** [review]

**Pull request merged**